### ISSUE TITLE
[Question]: Read/write file partially (random access to file) example?

### ISSUE THREAD

**Issue by isabsent** [hierynomus/smbj#691]
I have seen a random access to a file in your library, but can't find code sample with this action. Can you provide such examples?

**Comment by hierynomus** [comment]
I disagree, it is possible to open a file for write, you just need to ensure you close it first after opening it for read only. It is recommended to use a little privileges as possible. If reading/writing to a file, then open with both.

**Comment by hierynomus** [comment]
Please close the issue if it is solved for you

**Comment by isabsent** [comment]
> I disagree, it is possible to open a file for write, you just need to ensure you close it first after opening it for read only. It is recommended to use a little privileges as possible. If reading/writing to a file, then open with both.

Yes, but it is necessary to read and write the same file many times for using it as an encrypted container in my case and there is no sense to close it and reopen for a writing after each reading.

**Comment by hierynomus** [comment]
As I said, if your usecase is to read/write in a single transaction, open with both, else it's best practice to request as least permissions as needed.
